File: server/src/main/scala/kpn/api/common/changes/filter/ChangesFilterPeriod.scala
package kpn.api.common.changes.filter

case class ChangesFilterPeriod(
  name: Long,
  totalCount: Long,
  impactedCount: Long,
  current: Boolean = false,
  selected: Boolean = false,
  periods: Seq[ChangesFilterPeriod] = Seq.empty
)

File: backend/api/src/main/java/kpn/api/common/changes/filter/ChangesFilterPeriod.java
package kpn.api.common.changes.filter;

import com.google.common.collect.ImmutableList;

public record ChangesFilterPeriod(
  Long name,
  Long totalCount,
  Long impactedCount,
  Boolean current,
  Boolean selected,
  ImmutableList<ChangesFilterPeriod> periods
) {
}
